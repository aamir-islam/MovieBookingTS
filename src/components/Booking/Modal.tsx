import React from "react";
import {
  ModelStyle,
  ImageStyles,
  ModelDataFlex,
  ModelScreenStyle,
  HeadingStyle,
  Heading5Style,
  CloseButton,
  SeatHeading,
  HeadingSmallStyle,
  SeatNameStyle,
  SeatHeadingH3,
} from "../Styles/Modal.styles";
import { imgUrl } from "../constants/global";
import { useParams } from "react-router-dom";
import close from '../Assets/Xclose.png'
type Props = {
  SeatTempFun: (arr: string[]) => void;
  ModalVisibleFun: (arr: boolean) => void;
  seatArr: string[];
};


const Modal = ({
  SeatTempFun,
  ModalVisibleFun,
  seatArr,
}: Props) => {

  const state = useParams()
  
  const ImagePath = localStorage.getItem(state.title ||'{}')
  const total = seatArr.length * 250;
  const SGST = (total * 12) / 100;

  const modalClose = () => {
    SeatTempFun([]);
    ModalVisibleFun(false);
  };

  return (
    <ModelScreenStyle>  
    <ModelStyle>
      <CloseButton>
        <img onClick={modalClose} src={close} alt="close" />
      </CloseButton>
      <HeadingStyle>Thank you for booking</HeadingStyle>
      <Heading5Style>Order Summary</Heading5Style>
      <article>
        <ModelDataFlex>
          <article>
            <ImageStyles src={imgUrl+ImagePath} alt="movie" />
          </article>
          <article>
            <SeatHeadingH3>Seats:</SeatHeadingH3>
            <ModelDataFlex>
            <SeatNameStyle >{seatArr.toString()}</SeatNameStyle>  
            </ModelDataFlex>
            <ModelDataFlex>
              <article>
                <SeatHeading>{seatArr.length}*250:</SeatHeading>
                <HeadingSmallStyle>SGST(12%):</HeadingSmallStyle>
                <br />
                <HeadingSmallStyle>CGST(12%):</HeadingSmallStyle>
              </article>
              <article>
                <SeatHeading>{total}</SeatHeading>
                <HeadingSmallStyle>{SGST}</HeadingSmallStyle>
                <br />
                <HeadingSmallStyle>{SGST}</HeadingSmallStyle>
              </article>
            </ModelDataFlex>
            <hr />
            <ModelDataFlex>
              <SeatHeading>Total:</SeatHeading>
              <SeatHeading>{total + SGST * 2}</SeatHeading>
            </ModelDataFlex>
          </article>
        </ModelDataFlex>
      </article>
    </ModelStyle>
    </ModelScreenStyle>  
  );
};

export default Modal;
