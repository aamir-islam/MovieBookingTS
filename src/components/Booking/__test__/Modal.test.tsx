import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import Modal from "../Modal";

const data = ['A1', 'B1', 'C1', 'D1', 'A2', 'B2', 'C3']
const mockSetVisible = jest.fn();
const mockSeatSelector = jest.fn()



describe("Modal test", () => {
    test('should take a snapshot', () => {
        const { container } = render(<Modal SeatTempFun={mockSeatSelector} ModalVisibleFun={mockSetVisible} seatArr={data} />)
        expect(container).toMatchSnapshot()
    })

    test("booking modal SeatTempFu  function", () => {
        render(<Modal SeatTempFun={mockSeatSelector} ModalVisibleFun={mockSetVisible} seatArr={data} />)
        const numberList = screen.getByTestId<HTMLImageElement>("closeFunc")
        fireEvent.click(numberList);
        expect(mockSeatSelector).toHaveBeenCalled()
    })

    test("booking modal ModalVisibleFun", () => {
        render(<Modal SeatTempFun={mockSeatSelector} ModalVisibleFun={mockSetVisible} seatArr={data} />)
        const numberList = screen.getByTestId<HTMLImageElement>("closeFunc")
        fireEvent.click(numberList);
        expect(mockSetVisible).toHaveBeenCalled()

    })

})
