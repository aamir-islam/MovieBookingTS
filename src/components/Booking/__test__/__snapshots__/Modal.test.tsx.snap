// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal test should take a snapshot 1`] = `
<div>
  <article
    class="sc-bczRLJ dyGsPf"
  >
    <main
      class="sc-gsnTZi jcrqNa"
    >
      <button
        class="sc-eCYdqJ cCobBR"
        data-testid="closeFunc"
      >
        <img
          alt="close"
          src="Xclose.png"
        />
      </button>
      <h1
        class="sc-dkzDqf jHTDOg"
      >
        Thank you for booking
      </h1>
      <h5
        class="sc-hKMtZM kAGHsJ"
      >
        Order Summary
      </h5>
      <article>
        <article
          class="sc-gKXOVf frJWtB"
        >
          <article>
            <img
              alt="movie"
              class="sc-jSMfEi cvEyfM"
              src="https://image.tmdb.org/t/p/w500/null"
            />
          </article>
          <article>
            <h3
              class="sc-papXJ ilwbdE"
            >
              Seats:
            </h3>
            <article
              class="sc-gKXOVf frJWtB"
            >
              <h1
                class="sc-iqcoie dIqnFt"
              >
                A1,B1,C1,D1,A2,B2,C3
              </h1>
            </article>
            <article
              class="sc-gKXOVf frJWtB"
            >
              <article>
                <h5
                  class="sc-ftvSup gnwneK"
                >
                  7
                  *250:
                </h5>
                <small
                  class="sc-kDDrLX eRepAA"
                >
                  SGST(12%):
                </small>
                <br />
                <small
                  class="sc-kDDrLX eRepAA"
                >
                  CGST(12%):
                </small>
              </article>
              <article>
                <h5
                  class="sc-ftvSup gnwneK"
                >
                  1750
                </h5>
                <small
                  class="sc-kDDrLX eRepAA"
                >
                  210
                </small>
                <br />
                <small
                  class="sc-kDDrLX eRepAA"
                >
                  210
                </small>
              </article>
            </article>
            <hr />
            <article
              class="sc-gKXOVf frJWtB"
            >
              <h5
                class="sc-ftvSup gnwneK"
              >
                Total:
              </h5>
              <h5
                class="sc-ftvSup gnwneK"
              >
                2170
              </h5>
            </article>
          </article>
        </article>
      </article>
    </main>
  </article>
</div>
`;
